pep-0596.rst

```diff
@@ -0,0 +1,191 @@
+PEP: 596
+Title: Use UTF-8 for default text file encoding
+Author: Inada Naoki  <songofacandy@gmail.com>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 04-Jun-2019
+Python-Version: 3.9
+
+Abstract
+========
+
+Currently, ``TextIOWrapper`` uses ``locale.getpreferredencoding(False)``
+when ``encoding`` is not specified.
+
+This PEP proposes changing the default encoding to "UTF-8" regardless of
+platform or locale.
+
+
+Motivation
+==========
+
+People assume UTF-8 always
+--------------------------
+
+Package authors using macOS or Linux may forget that the default encoding
+is not always UTF-8.
+
+For example, ``long_description = open("README.md").read()`` in
+``setup.py`` is common mistake.  If there are at least one emoji or any
+other non-ASCII characters in the ``README.md`` file, many Windows users
+cannot install the package by ``UnicodeDecodeError``.
+
+
+Code page is not stable
+-----------------------
+
+Some tools on Windows change code page to 65001 (UTF-8), and Microsoft
+is using UTF-8 and cp65001 more widely in recent Windows 10.
+
+For example, "Command Prompt" uses legacy code page by default.
+But WSL changes the code page to 65001, and  ``python.exe`` on Windows
+can be executed from WSL.  So ``python.exe`` executed from legacy
+console and from WSL cannot read text files written by each other.
+
+Many Windows users don't understand code pages and which code page is
+currently used.  So changing default text file encoding based on current
+code page may cause confusion.
+
+Consistent default text encoding will make Python behavior more expectable
+and easy to learn.
+
+
+Use UTF-8 by default is easier to new programmers
+-------------------------------------------------
+
+Python is one of the most popular first programming language.
+
+New programmers may not know about encoding.  When they download text data
+written in UTF-8 from the internet, they are forced to know encoding.
+
+Popular text editers like VS Code or Atom use UTF-8 by default.
+Even notepad.exe uses UTF-8 by default from Windows 10 2019 may update.
+(Note that Python 3.9 will be released in 2021.)
+
+It would be nice if new programmers are not forced to know about encoding
+until they need to handle text files encoded in encoding other than UTF-8.
+
+
+Specification
+=============
+
+From Python 3.9, default encoding of ``TextIOWrapper`` and ``open()`` is
+changed from ``locale.getpreferredencoding(False)`` to "UTF-8".
+
+When there is device encoding (``os.device_encoding(buffer.fileno())``),
+itstill precedes than this default encoding.
+
+
+Not affected areas
+------------------
+
+Unlike UTF-8 mode, ``locale.getpreferredencoding(False)`` returns
+locale encoding.  Locale encoding is still respected when it makes sense.
```

Comment:
Can't we "fix" the UTF-8 Mode instead? I decided to modify locale.getpreferredencoding(False) because I was too lazy to fix all locations relying directly or implicitly on locale.getpreferredencoding(False). I don't recall the details, but they are way more locations than what I expected.

At the end, making locale.getpreferredencoding(False) to lie looks like a simpler solution to my problem. It makes Python more "consistent" in term of encodings. Even if I dislike making functions lie :-(

Reply:
I think users of UTF-8 mode will love this lie.  But I'm not sure this lie is good when UTF-8 mode become default.

Anyway, after this PEP is accepted and implemented, ``locale.getpreferredencoding(False)`` is not used in TextIOWrapper.  Checking all callers will be easier than before.